### PR TITLE
Amwal widget

### PR DESCRIPTION
**Pull request by ahmardi**

### CODE REVIEW

File: view/frontend/templates/express/widget-checkout-button.phtml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                    <?php endif; ?>
+                >
+                </div>
+                <script type="text/x-magento-init">
+                    {
+                        "#<?= $buttonId ?>": {
+                            "Amwal_Payments/js/checkout-button": {
+                                "buttonId" : "<?= $buttonId ?>",
+                                "triggerContext": "<?= $triggerContext ?>",
+                                "hideProceedToCheckout": "true"
```
Comment (sameh-amwal) [inline view/frontend/templates/express/widget-checkout-button.phtml:111]: is `hideProceedToCheckout` needed here?

File: view/frontend/web/js/checkout-button.js [hunk 2]
```diff
@@ -139,6 +143,69 @@ function ($, Component, placeAmwalOrder, payAmwalOrder, amwalErrorHandler, urlBu
                 attributes: false,
                 characterData: false
             });
+        },
+
+        /**
+         * initializeAmwalWidget
+         */
+
+        initializeAmwalWidget: function () {
```
Comment (sameh-amwal) [inline view/frontend/web/js/checkout-button.js:152]: I see a lot of repeated code. Can the code be refactored and shared with the `initializeProductDetail` function?

File: view/frontend/templates/express/widget-checkout-button.phtml (hunk 1 above)
Comment (ahmardi) [reply view/frontend/templates/express/widget-checkout-button.phtml:111]: No, i can remove it

File: view/frontend/web/js/checkout-button.js (hunk 2 above)
Comment (ahmardi) [reply view/frontend/web/js/checkout-button.js:152]: Yes, i will mere the `AmwalWidget` logic with `initializeProductDetail`

**Review by sameh-amwal — APPROVED** [review]

**Pull request merged**